bundles/binding/org.openhab.binding.expire/src/main/java/org/openhab/binding/expire/internal/ExpireBinding.java

```diff
@@ -0,0 +1,152 @@
+     * was called.
+     */
+    private BundleContext bundleContext;
+
+    /**
+     * The refresh interval is used to check if any of the bound items is expired.
+     * One second should be fine for all use cases, so it's final and cannot be configured.
+     */
+    private final long refreshInterval = 1000;
+
+    public ExpireBinding() {
+    }
+
+    /**
+     * Called by the SCR to activate the component with its configuration read from CAS
+     *
+     * @param bundleContext BundleContext of the Bundle that defines this component
+     * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+     */
+    public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+        this.bundleContext = bundleContext;
+
+        setProperlyConfigured(true);
+    }
+
+    /**
+     * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+     *
+     * @param configuration Updated configuration properties
+     */
+    public void modified(final Map<String, Object> configuration) {
+        // update the internal configuration accordingly
+    }
+
+    /**
+     * Called by the SCR to deactivate the component when either the configuration is removed or
+     * mandatory references are no longer satisfied or the component has simply been stopped.
+     *
+     * @param reason Reason code for the deactivation:<br>
+     *            <ul>
+     *            <li>0 – Unspecified
+     *            <li>1 – The component was disabled
+     *            <li>2 – A reference became unsatisfied
+     *            <li>3 – A configuration was changed
+     *            <li>4 – A configuration was deleted
+     *            <li>5 – The component was disposed
+     *            <li>6 – The bundle was stopped
+     *            </ul>
+     */
+    public void deactivate(final int reason) {
+        this.bundleContext = null;
+        // deallocate resources here that are no longer needed and
+        // should be reset when activating this binding again
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected long getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected String getName() {
+        return "Expire Refresh Service";
+    }
+
+    protected static class ExpireConfig {
+        protected long expireAfterMs;
+
+        protected long nextExpireTs;
+    }
+
+    protected static Map<String, Long> nextExpireTsMap = new HashMap<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void execute() {
+        for (ExpireBindingProvider provider : providers) {
+            for (String itemName : provider.getItemNames()) {
+                Long nextExpireTs = nextExpireTsMap.get(itemName);
+                if (nextExpireTs == null || nextExpireTs <= System.currentTimeMillis()) {
```

Comment:
It depends on how initial state restores are handled. If it's sure that the restore is passed as state update to the binding, all is fine with this implementation. If not, @watou 's suggetion won't be sufficient because the restored value would never expire.

I changed it the following way: if an item has no entry in the expireTsMap, I create one with the initial expire time. So if ther's no update after initialisation, the item is expired after expire time instead and not immediatelly. It it would receive an update at start, this will work as well.